### PR TITLE
added math symbols and some standard data types for python evaluation

### PR DESCRIPTION
**Pull request by rhaschke**
... as discussed in https://github.com/ros/ros_comm/pull/784#issuecomment-209135252

### CODE REVIEW

File: tools/roslaunch/src/roslaunch/substitution_args.py [hunk 1]
```diff
@@ -279,8 +280,11 @@ def _arg(resolved, a, args, context):
 # context.  We disable all the builtins, then add back True and False, and also
 # add true and false for convenience (because we accept those lower-case strings
 # as boolean values in XML).
-_eval_dict={'true': True, 'false': False, 'True': True, 'False': False, '__builtins__': {},
+_eval_dict={'true': True, 'false': False, 'True': True, 'False': False,
+            '__builtins__': {k: __builtins__[k] for k in ['list', 'dict', 'map', 'str', 'float', 'int']},
             'env': _eval_env, 'optenv': _eval_optenv, 'find': _eval_find}
```
Comment (dirk-thomas) [inline tools/roslaunch/src/roslaunch/substitution_args.py]: This block becomes more difficult to read as it grows. Can you update it to:

```
eval_dict = {
    'true': True, 'false': False,
    'True': True, 'False': False,
    '__builtins__': {k: __builtins__[k] for k in ['dict', 'float', 'int', 'list', 'map', 'str']},
    'env': _eval_env, 'optenv': _eval_optenv,
    'find': _eval_find
}
```

**Comment by dirk-thomas** [comment]
One test failed.

**Comment by rhaschke** [comment]
The failed tests are unrelated to roslaunch:

```
throttle: /usr/include/boost/thread/pthread/recursive_mutex.hpp:113: void boost::recursive_mutex::lock(): Assertion `!pthread_mutex_lock(&m)' failed.
```

**Comment by dirk-thomas** [comment]
I was referring to the other one: http://build.ros.org/job/Kpr__ros_comm__ubuntu_xenial_amd64/61/testReport/junit/unit/test_substitution_args/test_resolve_args/

**Comment by rhaschke** [comment]
Oops. I missed that one. Hopefully fixed now. And squashed ;-)

**Comment by dirk-thomas** [comment]
Thank you!

**Pull request merged**